### ISSUE TITLE
Introduce warning against embedded images for representing kanji characters

### ISSUE THREAD

**Issue by murata2makoto** [w3c/epub-specs#919]
Japanese EPUB publications often contain images that mimic kanji characters.  Such images significantly hamper accessibility.   I propose to provide some warning against such embedded images in EPUB Accessibility 1.0.   (I am sure there will be some oppositions to this proposal.)

Such images can almost always be replaced by characters in Unicode.  But publishers are often unaware of characters beyond JIS X 0208.  It is an old standard having a relatively small number of characters.

Such images are rarely used on the Web.  Thus, WCAG20 does not have anything specific to such embedded images. 

The EBPAJ EPUB guideline recommends the use of such embedded images when characters  beyond JIS X 0213 (which is a newer standard having more characters) are needed.

Ideographic variation selectors of Unicode will hopefully address this issue completely, but they are not yet supported by EPUB reading systems.

**Comment by mattgarrish** [comment]
Not sure I understand the use case. If the characters have to be represented as images because there's no other way to represent them, then how can we discourage people from using them? There needs to be a solution of some kind.

Are they using img, which can be made accessible with alt text, or is this a case of private use characters?

All I can really say is if people are using images inaccessibly where they should be using text, then that's already a violation of 1.3.1 and 1.4.5, so such publications wouldn't pass.

If you just want to recommend using the proper unicode characters whenever available, that could be done as a technique that is broadly applicable, but using Japanese as an example.

**Comment by murata2makoto** [comment]
It is almost always possible to represent characters as Unicode characters.  But people tend to use embedded images even if they don't have to.

You think that this is just a special case of  1.3.1 and 1.4.5.  Could you give me a pointer?  I searched for "1.3.1" in EPUB Accessibility Techniques 1.0 and EPUB Accessibility 1.0, but cannot find it.

**Comment by mattgarrish** [comment]
It's a failure of the WCAG criteria if they're images in place of text, as it would represent a loss of information. 1.3.1 covers this,[1] and 1.4.5 sets out rules for images of text.[2]

If it's commonly done, we could still point out in a technique, and recommend unicode as a means of complying.

[1] https://www.w3.org/TR/WCAG20/#content-structure-separation-programmatic
[2] https://www.w3.org/TR/WCAG20/#visual-audio-contrast-text-presentation

**Comment by murata2makoto** [comment]
Thanks.  Yes,  addition as a technique would be great, since this 
issue is significant in Japanese EPUB.  I can imagine that this 
problem arises in Asia.

**Comment by mattgarrish** [comment]
Does this capture the issue correctly?

> 4.5 Text
> TEXT-001 - Use Unicode for text content
> [WCAG 2.0] Success Criterion 1.4.5 governs the acceptable uses of images in place of text. It is still a common practice in digital publications to use images of text despite the availability of Unicode character equivalents, however. For example, Japanese publishers often use images of characters that are not available in JIS X 0213 for font compatibility reasons.
> 
> Such practice is not conformant with WCAG unless an alternative text value is provided with the equivalent information. Even in such cases where an alternative can be provided, the practice is strongly discouraged if the image will negatively affect text-to-speech playback (e.g., if the image is one character in a word).
> 
> Authors are recommended to use Unicode characters for all text content, and reserve the use of images of text for the cases spelled out in the WCAG criterion.

**Comment by murata2makoto** [comment]
I posted this issue in a Japanese EPUB forum.  Some publisher is quite angry with me.  Indeed, there are some inevitable cases.  Nevertheless, I still think that some warning is very desirable. 

I think that the third paragraph has to be softened, since this is just AA rather than A.

**Comment by mattgarrish** [comment]
But having text is core to WCAG. I pointed to 1.3.1 above, but what I wanted was 1.1.1, which states:

1.1.1 Non-text Content: All non-text content that is presented to the user has a text alternative that serves the equivalent purpose, except for the situations listed below. (Level A)

The techniques are not normative, so ultimately people can make of them what they will.

When it comes to stating conformance to WCAG, though, there has to be proof that whatever the author has done has retained the text requirement. A publication with images instead of text wouldn't meet that bar. Similarly, a publication that can't be read because words are part text and part image would also fail.

**Comment by murata2makoto** [comment]
Re: For example, Japanese publishers often use images of characters that are not available in JIS X 0213 for font compatibility reasons.

What is most problematic is the use of images of characters even when JIS X 0213 contain equivalent characters.   Such use is not uncommon and can hopefully be addressed soon.  I thus would like to mention such use rather than characters beyond JIS X 0213.

How about "For example, some Japanese EPUB publications use images of characters even when equivalent characters exist in JIS X 0213."?

**Comment by mattgarrish** [comment]
Okay, I wasn't sure if I'd captured it correctly. We'll need to run this past the group, as the review period I believe is ending soon. It is only an informational document, but it would be good to have everything done at the same time. I'll send an email to the list.

**Comment by murata2makoto** [comment]
Somebody thinks that " It is still a common practice in digital publications to use images of text despite the availability of Unicode character equivalents, however" implies that we should not use images of text if there are Unicode equivalents.   But we cannot use Unicode characters "If for any reason, the author cannot format the text to get the same effect, the effect won't be reliably presented on the commonly available user agents, or using a technology to meet this criterion would interfere with meeting other criteria such as 1.4.4" (WCAG 2.0).

**Comment by mattgarrish** [comment]
What "effect" necessitates images, though? That sentence refers to stylized text, replicas, etc. as listed in the examples of where it is necessary to use text.

And even in the cases listed, there still has to be an accessible text equivalent; the requirement doesn't go away.

As I said above, if the publication is still fully accessible despite having images of text, then it can pass WCAG Level A, but that also means ensuring that the images don't affect the readability. When you get down to the level of using images for individual characters, I'm suspect about the actual accessibility that can be achieved. You'd need to test with actual users, at a minimum.

But if you want another opinion on this, you might want to pose this problem to the [WAI interest group mailing list](https://lists.w3.org/Archives/Public/w3c-wai-ig/). That's a good forum for accessibility questions like this.

**Comment by murata2makoto** [comment]
I propose to add "There are some plausible reasons for this practice" at the end of the first paragraph, and add "However, the use of characters is central to accessibility, as shown in WGCA 2.0" at the beginning of the second paragraph.   I think that the change to the first para acknowledges inevitable cases and that the change to the second para makes the goal clearer.

**Comment by mattgarrish** [comment]
I've been trying to rewrite the section to reference the Level A requirement and tone down that we are recommending a practice that we're just pointing out (i.e., this isn't a practice we're placing on top of wcag).

See if this is more palatable: https://rawgit.com/IDPF/epub-revision/master/build/a11y/techniques/techniques.html#text-001
